js/fetch.ts

```diff
@@ -410,28 +420,58 @@ export async function fetch(
     }
   }
 
-  // Send Fetch message
-  const builder = flatbuffers.createBuilder();
-  const headerOff = msgHttpRequest(builder, url, method, headers);
-  const resBase = await sendAsync(
-    builder,
-    msg.Any.Fetch,
-    msg.Fetch.createFetch(builder, headerOff),
-    body
-  );
-
-  // Decode FetchRes
-  assert(msg.Any.FetchRes === resBase.innerType());
-  const inner = new msg.FetchRes();
-  assert(resBase.inner(inner) != null);
-
-  const header = inner.header()!;
-  const bodyRid = inner.bodyRid();
-  assert(!header.isRequest());
-  const status = header.status();
-
-  const headersList = deserializeHeaderFields(header);
-
-  const response = new Response(status, headersList, bodyRid);
-  return response;
+  while (true) {
```

Comment:
As @bartlomieju mentioned, there should be some counter here to avoid infinite looping. 

We should research how other software handles this. I'm not familiar with this. If you want to put that off until later, that's fine, just add a TODO and set it to a maximum of, say, 3 redirects.